journal/src/main/scala/com/evolutiongaming/kafka/journal/Events.scala

```diff
@@ -28,7 +31,9 @@ object Events {
     codecs.choice(version1, version0, default)
   }
 
-  implicit def eventsToBytes[F[_] : FromAttempt]: ToBytes[F, Events] = ToBytes.fromEncoder
+  implicit def eventsToBytes[F[_] : FromAttempt](implicit jsonCodec: JsonCodec[Try]): ToBytes[F, Events] =
+    ToBytes.fromEncoder
 
-  implicit def eventsFromBytes[F[_] : FromAttempt]: FromBytes[F, Events] = FromBytes.fromDecoder
+  implicit def eventsFromBytes[F[_] : FromAttempt](implicit jsonCodec: JsonCodec[Try]): FromBytes[F, Events] =
```

Comment:
Why whole `JsonCodec[Try]` is required here, rather than `JsonCodec.Decode[Try]`?

Reply:
It uses `FromBytes.fromDecoder` which requires `scodec.Decoder`. The decoder instance is provided by `scodec.Codec` which is in turn requires both `Encode` and `Decode` i.e `JsonCodec`. To fix this we would have to split codec into encoder and decoder.

Reply:
Do you mean `Codec[Events]`? I think it is an excellent idea to split it.